content/ADR-138-extended-passport-support.md

```diff
@@ -0,0 +1,73 @@
+---
+adr: 138
+date: 2022-11-25
+title: Technical Assessment for the Extended Passport
+authors:
+  - manumena
+status: Draft
+type: RFC
+spdx-license: CC0-1.0
+redirect_from:
+  - /rfc/RFC-1
+---
+
+## Need
+
+A New Passport view is being developed by the Explorer.
+This view will require storing information in the profile entity and retrieving things as collectibles to be displayed, as well as being able to block, add as a friend or initiate a chat.
+
+## Requirements
+The applicable requirements from the Catalyst point of view:
+
+1. Intro of a user:
+Use the "description" field from profiles to use in the "Intro" section or create a new one if it is different from the "description".
+
+2. Block/Unblock users:
+Use the "blocked" field from profiles to see/report blocked users.
+
+3. Equipped wearables:
+Take the information from the profile. There are the equipped wearables, emotes and the current name.
+
+4. Items: wearables/emotes/names/lands:
+For the wearables and emotes there are two available endpoints `/wearables-by-owner` and `/emotes-by-owner` that can be used for that. It will be necessary to create `/names-by-owner` and `/lands-by-owner` endpoints with similar behavior.
+
+5. Guest: Passport for guest users will show only name and description, there is nothing to do for guests from the catalyst end.
+
+## Performance
+To load a passport, first, there will be made 5 requests to a Catalyst (lambdas):
+
+- `/profiles`
+- `/wearables-by-owner`
+- `/emotes-by-owner`
+- `/lands-by-owner`
+- `/names-by-owner`
+
+Once this information is obtained, to display wearables and emotes it will be needed a couple of requests per each emote/wearable to download its image/thumbnail to be shown. This does not scale because some users have more than 1k NFTs. So the explorer should implement a lazy loading mechanism (for image/thumbnail) as the user browses the collectibles.
```

Comment:
```suggestion
Once this information is obtained, to display wearables and emotes a few extra requests will be needed for each emote/wearable to download their image/thumbnail. This does not scale because some users have more than 1k NFTs. So the Explorer should implement a lazy loading mechanism (for image/thumbnail) as the user browses the collectibles.
```